app/src/main/java/org/fossasia/susi/ai/Login/LoginActivity.kt

```diff
@@ -0,0 +1,165 @@
+    lateinit var email: TextInputLayout
+    @BindView(R.id.email_input)
+    lateinit var autoCompleteEmail: AutoCompleteTextView
+    @BindView(R.id.password)
+    lateinit var password: TextInputLayout
+    @BindView(R.id.log_in)
+    lateinit var logIn: Button
+    @BindView(R.id.susi_default)
+    lateinit var susiServer: RadioButton
+    @BindView(R.id.personal_server)
+    lateinit var personalServer: RadioButton
+    @BindView(R.id.input_url)
+    lateinit var url: TextInputLayout
+
+    var loginPresenter: ILoginPresenter? = null
+    var progressDialog: ProgressDialog? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        ButterKnife.bind(this)
+
+        loginPresenter = LoginPresenter()
+        loginPresenter!!.onAttach(this)
+
+    }
+
+    override fun onLoginSuccess() {
+        val intent = Intent(this@LoginActivity, MainActivity::class.java)
+        intent.putExtra(Constant.FIRST_TIME, true)
+        startActivity(intent)
+        finish()
+    }
+
+    override fun alreadyLoggedIn() {
+        val intent = Intent(this@LoginActivity, MainActivity::class.java)
+        intent.putExtra(Constant.FIRST_TIME, false)
+        startActivity(intent)
+        finish()
+    }
+
+    override fun skipLogin() {
+        val intent = Intent(this@LoginActivity, MainActivity::class.java)
+        intent.putExtra(Constant.FIRST_TIME, false)
+        startActivity(intent)
+        finish()
+    }
+
+    override fun incorrectEmailError() {
+        val builder = AlertDialog.Builder(this@LoginActivity)
+        builder.setTitle(R.string.email_invalid_title)
+        builder.setMessage(R.string.email_invalid)
+                .setCancelable(false)
+                .setPositiveButton(applicationContext.getString(R.string.retry), null)
+        val alert = builder.create()
+        alert.show()
+        val ok = alert.getButton(DialogInterface.BUTTON_POSITIVE)
+        ok.setTextColor(Color.RED)
+    }
+
+    override fun emptyEmailError() {
+        email.setError(getString(R.string.email_cannot_be_empty));
+    }
+
+    override fun emptyPasswordError() {
+        password.setError(getString(R.string.password_cannot_be_empty));
+    }
+
+    override fun emptyURLError() {
+        url.setError(getString(R.string.url_cannot_be_empty));
+    }
+
+    override fun invalidURLError() {
+        url.setError(getString(R.string.invalid_url))
+    }
+
+    override fun showProgress() {
+        progressDialog!!.show()
+    }
+
+    override fun hideProgress() {
+        progressDialog!!.hide()
+    }
+
+    @OnClick(R.id.personal_server)
+    fun showURL() {
+        url.visibility = View.VISIBLE
+    }
+
+    @OnClick(R.id.susi_default)
+    fun hideURL() {
+        url.visibility = View.GONE
+    }
+
+    @OnClick(R.id.sign_up)
+    fun signUp() {
+        val intent = Intent(this@LoginActivity, SignUpActivity::class.java)
```

Comment:
There are many more uses of Anko. We will proceed with them once the shifting to kotlin is complete. It will cause confusion if we do it now.